src/main/java/com/aws/greengrass/clientdevices/auth/certificate/CertificateStore.java

```diff
@@ -139,6 +144,15 @@ public X509Certificate getCACertificate() throws KeyStoreException {
         return (X509Certificate) getCaCertificateChain()[0];
     }
 
+    /**
+     * Sets the CA certificate chain.
+     *
+     * @param caCertificateChain Array of CA certificates
+     */
+    public void setCaCertificateChain(Certificate... caCertificateChain) {
+        this.caCertificateChain = caCertificateChain;
+        eventEmitter.emit(new ConfiguredCertificateAuthorityEvent(caCertificateChain));
```

Comment:
This is beautiful

Reply:
The event name is not very expressive of what we are doing. Maybe `CACertificateChainChanged`.